lib/components/component.dart

```diff
@@ -15,6 +15,30 @@ import '../extensions/vector2.dart';
 abstract class Component {
   final EffectsHandler _effectsHandler = EffectsHandler();
 
+  /// Whether this component has been loaded yet. If not loaded, [BaseGame] will not try to render it.
+  ///
+  /// Sprite based components can use this to let [BaseGame] know not to try to render when the [Sprite] has not been loaded yet.
+  /// Note that for a more consistent experience, you can pre-load all your assets beforehand with Flame.images.loadAll.
+  bool loaded = true;
+
+  /// Whether this component is HUD object or not.
+  ///
+  /// HUD objects ignore the [BaseGame.camera] when rendered (so their position coordinates are considered relative to the device screen).
+  bool isHud = false;
+
+  /// Render priority of this component. This allows you to control the order in which your components are rendered.
+  ///
+  /// Components are always updated and rendered in the order defined by what this number is when the component is added to the game.
+  /// The smaller the priority, the sooner your component will be updated/rendered.
+  /// It can be any integer (negative, zero, or positive).
+  /// If two components share the same priority, they will probably be drawn in the order they were added.
+  final int priority = 0;
```

Comment:
I think this can't be initialized here right? This should be received on the constructor, otherwise this will be always zero, on all components.

Or am I missing something?

Reply:
I was thinking that it should be overridden by subclasses, but it is probably better to have it in the constructor like you say

Reply:
Just pushed a change with it.

Reply:
Fair enough! I think it makes sense, overriding this will be the easier way, adding on the constructor is a good addition, but I think it could be done on a different PR maybe